mmv1/third_party/terraform/provider/provider_project_test.go

```diff
@@ -0,0 +1,257 @@
+					// unset
+					t.Setenv("GOOGLE_PROJECT", "")
+					t.Setenv("GOOGLE_CLOUD_PROJECT", "")
+					// set
+					t.Setenv("GCLOUD_PROJECT", "GCLOUD_PROJECT") // used
+					t.Setenv("CLOUDSDK_CORE_PROJECT", "CLOUDSDK_CORE_PROJECT")
+				},
+				Config: testAccSdkProvider_projectInEnvsOnly(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_sdk.default", "project", "GCLOUD_PROJECT"),
+				),
+			},
+			{
+				// CLOUDSDK_CORE_PROJECT is used 4th if set
+				PreConfig: func() {
+					// unset
+					t.Setenv("GOOGLE_PROJECT", "")
+					t.Setenv("GOOGLE_CLOUD_PROJECT", "")
+					t.Setenv("GCLOUD_PROJECT", "")
+					// set
+					t.Setenv("CLOUDSDK_CORE_PROJECT", "CLOUDSDK_CORE_PROJECT")
+				},
+				Config: testAccSdkProvider_projectInEnvsOnly(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_sdk.default", "project", "CLOUDSDK_CORE_PROJECT"),
+				),
+			},
+		},
+	})
+}
+
+func testAccSdkProvider_project_emptyStringValidation(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	context := map[string]interface{}{
+		"project": "",
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccSdkProvider_projectInProviderBlock(context),
+				PlanOnly:    true,
+				ExpectError: regexp.MustCompile("expected a non-empty string"),
+			},
+		},
+	})
+}
+
+func testAccSdkProvider_project_unknownHandling(t *testing.T) {
+
+	project := envvar.GetTestProjectFromEnv()
+	context := map[string]interface{}{
+		"org_id":             envvar.GetTestOrgFromEnv(t),
+		"billing_account_id": envvar.GetTestBillingAccountFromEnv(t),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"random": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSdkProvider_projectUnknownHandling(context),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Matches ENV instead of the project id output from google_project resource
+					resource.TestCheckResourceAttr("data.google_provider_config_sdk.default", "project", project),
+				),
+			},
+			{
+				// Unset all ENVs for project
+				PreConfig: func() {
+					for _, v := range envvar.ProjectEnvVars {
+						t.Setenv(v, "")
+					}
+				},
+				Config: testAccSdkProvider_projectUnknownHandling(context),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Matches project id output from google_project resource
+					resource.TestMatchResourceAttr("data.google_provider_config_sdk.default", "project", regexp.MustCompile(fmt.Sprintf("tf-test-[0-9a-z]{16}|%s", project))),
```

Comment:
There's some flakiness I see when running this locally vs on the PR

Reply:
What kind of flakiness do you see?
Does the test fail locally?

Reply:
The flakiness was that locally I saw `tf-test-XXXXXX` was used here, but when I pushed to GitHub the tests running in the CI would have the value coming from environment variables. That's why the regex here can accept two alternate outcomes.

We'll have to see whether the test changes in 54eacde pass on this PR or not!